swapi/actions.js

```diff
@@ -35,10 +57,78 @@ export const onChooseEndpoint = (endpoint) => (dispatch, getState) => {
 
 	}
 
-	return loadData('https://swapi.co/api/' + endpoint + '/')
+	function loadSchema(url) {
+		dispatch({ type: types.START_LOAD });
+
+		return fetch(url + 'schema')
+			.then((response) => response.json())
+			.then((json) => {
+				dispatch({
+					type: types.SET_SCHEMA,
+					payload: {
+						endpoint: endpoint,
+						data: json,
+					}
+				});
+
+				dispatch({ type: types.DONE_LOAD });
+			})
+			.catch(() => console.error('Oops'))
+	}
+
+	if (isCached())
+		return;
+
+	return loadData(url)
+		.catch((err) => {
+			console.error(err);
+			alert('It went wrong.')
+		})
+		.then(() => loadSchema(url))
 		.catch((err) => {
 			console.error(err);
 			alert('It went wrong.')
 		})
 		.then(() => dispatch({ type: types.DONE_LOAD }))
 };
+
+export const onOrderList = (selectedKey) => (dispatch, getState) => {
+	dispatch({ type: types.START_LOAD });
+
+	var current_state = getState();
+
+	var data = [].concat(current_state.data[current_state.endpoint]);
+	data.sort((a, b) => (a[selectedKey].toString()).localeCompare(b[selectedKey].toString()));
```

Comment:
Der mangler lige at argument til `localeCompare` som angiver hvad for en locale du gerne vil bruge.